### ISSUE TITLE
Improving computational time  for the interactive case

### ISSUE THREAD

**Issue by copperwire** [copperwire/comp_phys_2#10]
So far computation of the interactive term has been done rather naively with updating the entire correlation term for each new position. This is wildly inefficient. Since each term in the correlation function is dependent only on the particle positions in question we can either: 

1. Do the metropolis sampling by way of moving *every* particle such that recalculating the correlation function is necessary (I don't know that this is a good idea)
2. Update the methods such that the product of jastrow factors is  only updated with the relevant term. 

Suggestion `1` requires little to no change in implementation - as only the proposed sampling `R_p` needs be changed by two lines. Suggestion `2` requires a bit more thought as we don't have an implementation of taking indices to the evaluation of the local energy. It might be simplest to just pass dummy variables to all methods, i.e the call to `psi_t -> E_l(R, i, j) ` for all wavefunctions and hamiltonians, but the indices are in effect dummy variables  for the non-interactive case. 

This issue depends on #2 and #7

**Comment by KariEriksen** [comment]
You might have to explain this problem to me a bit more in detail. Cause I'm not sure if I agree with you that the correlation term must not be updated at every new movement in position. But we can look at it later!

**Comment by copperwire** [comment]
So the implementation of the correlation wavefunction in the commit 95bdc3756d75457fd9ec7e9d0f18cd65f61742af in the file `gaussian_inter_numeric.cpp` implementes the suggested fix 2. as mentioned above. 

Let the strictly upper triangular matrix <a href="https://www.codecogs.com/eqnedit.php?latex=D" target="_blank"><img src="https://latex.codecogs.com/gif.latex?D" title="D" /></a>   be the matrix containing the distances between each particle pair as defined in the project. Then if you've moved particle <a href="https://www.codecogs.com/eqnedit.php?latex=k" target="_blank"><img src="https://latex.codecogs.com/gif.latex?k" title="k" /></a> then the update rule for <a href="https://www.codecogs.com/eqnedit.php?latex=D" target="_blank"><img src="https://latex.codecogs.com/gif.latex?D" title="D" /></a> should read: 

```
i = 0
while i < k :
    if r_ik > a: 
         D(i, k) = r_ik
    i ++
j = k
while j < N :
    if r_kj > a:
       D(k, j) = r_jk
    j++ 
```
Where `r_ij` is the distance between particle `i` and  `j` and `N` is the number of particles.

**Comment by copperwire** [comment]
This issue should then be closed together with #9 that @ulvik  is working on today.

**Comment by copperwire** [comment]
Not important, as it turns out
